file-source/src/test/java/org/springframework/cloud/stream/module/file/source/FileSourceTests.java

```diff
@@ -0,0 +1,196 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.file.source;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import org.hamcrest.Matchers;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.IntegrationTest;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.cloud.stream.test.binder.MessageCollector;
+import org.springframework.integration.file.splitter.FileSplitter;
+import org.springframework.messaging.Message;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+/**
+ * @author Gary Russell
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = FileSourceApplication.class)
+@DirtiesContext
+public abstract class FileSourceTests {
+
+	private static final String TMPDIR = System.getProperty("java.io.tmpdir");
+
+	private static final String ROOT_DIR = TMPDIR + File.separator + "dataflow-tests"
+			+ File.separator + "input";
+
+	protected File atomicFileCreate(String filename) throws FileNotFoundException, IOException {
+		File file = new File(ROOT_DIR, filename + ".tmp");
+		File fileFinal = new File(ROOT_DIR, "test.txt");
+		file.delete();
+		file.deleteOnExit();
+		fileFinal.delete();
+		fileFinal.deleteOnExit();
+		FileOutputStream fos = new FileOutputStream(file);
+		fos.write("this is a test\nline2\n".getBytes());
+		fos.close();
+		assertTrue(file.renameTo(fileFinal));
+		return fileFinal;
+	}
+
+	@Autowired
+	protected Source source;
+
+	@Autowired
+	protected MessageCollector messageCollector;
+
+	@IntegrationTest({ "directory = ${java.io.tmpdir}${file.separator}dataflow-tests${file.separator}input",
+		"fixedDelay = 100", "timeUnit = MILLISECONDS" })
+	public static class ContentPayloadTests extends FileSourceTests {
+
+		@Test
+		public void testSimpleFile() throws Exception {
+			String filename = "test.txt";
+			File file = atomicFileCreate(filename);
+			Message<?> received = messageCollector.forChannel(source.output()).poll(10, TimeUnit.SECONDS);
+			assertNotNull(received);
```

Comment:
I think the returned **message** can never be `null` (same can't be said of payload variant methods)